app/controllers/application_controller.rb

```diff
@@ -23,7 +25,7 @@ def previous_url_for_cookies_page
   end
 
   def after_sign_in_path_for(user)
-    stored_location_for(user) || helpers.profile_dashboard_path(user)
```

Comment:
This is a bit tenuous but `user` in this context was always the impersonator. It didn't break anything so I rolled with it.

Reply:
Thinking about it perhaps I need to set `pundit_user` in the application controller like I did in the impersonation one.

```ruby
def pundit_user
  true_user
end
```

Will give it a go.